modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -21,9 +21,14 @@ func GenerateAddress(moduleAccAddr sdk.AccAddress, portID string) sdk.AccAddress
 	return sdk.AccAddress(sdkaddress.Derive(moduleAccAddr, []byte(portID)))
 }
 
-// ParseAddressFromVersion trims the interchainaccounts version prefix and returns the associated account address
+// ParseAddressFromVersion attempts to extract the associated account address from the provided version string
 func ParseAddressFromVersion(version string) string {
```

Comment:
It might be best to explicitly return an error. Otherwise we need something like:
```go
if address == "" {
    return sdkerrors.Wrap(types.ErrInvalidVersion, "could not parse version")
}
```
in most places we call this function

Reply:
Agreed, will update